compiler/semtypinst.nim

```diff
@@ -231,6 +220,16 @@ proc replaceTypeVarsS(cl: var TReplTypeVars, s: PSym): PSym =
   # symbol is not our business:
   if cl.owner != nil and s.owner != cl.owner:
     return s
+
+  # XXX: Bound symbols in default parameter expressions may reach here.
+  # We cannot process them, becase `sym.n` may point to a proc body with
+  # cyclic references that will lead to an infinite recursion. Perhaps we
+  # should not use a black-list here, but a whitelist instead.
```

Comment:
How would this whitelist look like?

Reply:
Haven't experimented with this yet, but perhaps it needs to look only at `skGenericParam` and `skType` symbols.